csharp/src/Microsoft.ML.OnnxRuntime/OrtTypeInfo.shared.cs

```diff
@@ -0,0 +1,342 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+
+using Microsoft.ML.OnnxRuntime.Tensors;
+using System;
+using System.Runtime.InteropServices;
+
+namespace Microsoft.ML.OnnxRuntime
+{
+    /// <summary>
+    /// This class retrieves Type Information for input/outputs of the model.
+    /// </summary>
+    public class OrtTypeInfo : SafeHandle
+    {
+        internal OrtTypeInfo(IntPtr handle) : base(handle, true)
+        {
+        }
+
+        internal IntPtr Handle { get { return handle; } }
+
+        /// <summary>
+        /// Represents OnnxValueType of the OrtTypeInfo
+        /// </summary>
+        /// <value>OnnxValueType</value>
+        public OnnxValueType OnnxType
+        {
+            get
+            {
+                NativeApiStatus.VerifySuccess(NativeMethods.OrtGetOnnxTypeFromTypeInfo(handle, out IntPtr type));
```

Comment:
Can we cache the info returned from the native calls here? I can't think of a scenario where the type/shape info is going to change on the native side.